libraries/AP_AccelCal/AccelCalibrator.cpp

```diff
@@ -0,0 +1,477 @@
+    if (_status != ACCEL_CAL_COLLECTING_SAMPLE) {
+        return;
+    }
+
+    if (_samples_collected >= _conf_num_samples) {
+        set_status(ACCEL_CAL_FAILED);
+        return;
+    }
+
+    _sample_buffer[_samples_collected].delta_velocity += delta_velocity;
+    _sample_buffer[_samples_collected].delta_time += dt;
+
+    _last_samp_frag_collected_ms = AP_HAL::millis();
+
+    if (_sample_buffer[_samples_collected].delta_time > _conf_sample_time) {
+        Vector3f sample = _sample_buffer[_samples_collected].delta_velocity/_sample_buffer[_samples_collected].delta_time;
+        if (!accept_sample(sample)) {
+            set_status(ACCEL_CAL_FAILED);
+            return;
+        }
+
+        _samples_collected++;
+
+        if (_samples_collected >= _conf_num_samples) {
+            run_fit(MAX_ITERATIONS, _fitness);
+
+            if (_fitness < _conf_tolerance && accept_result()) {
+                set_status(ACCEL_CAL_SUCCESS);
+            } else {
+                set_status(ACCEL_CAL_FAILED);
+            }
+        } else {
+            set_status(ACCEL_CAL_WAITING_FOR_ORIENTATION);
+        }
+    }
+}
+
+// determines if the result is acceptable
+bool AccelCalibrator::accept_result() const {
+    if (fabsf(_param_struct.offset.x) > GRAVITY_MSS ||
+        fabsf(_param_struct.offset.y) > GRAVITY_MSS ||
+        fabsf(_param_struct.offset.z) > GRAVITY_MSS ||
+        _param_struct.diag.x < 0.8f || _param_struct.diag.x > 1.2f ||
+        _param_struct.diag.y < 0.8f || _param_struct.diag.y > 1.2f ||
+        _param_struct.diag.z < 0.8f || _param_struct.diag.z > 1.2f) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+// interface for LSq estimator to read sample buffer sent after conversion from delta velocity
+// to averaged acc over time
+bool AccelCalibrator::get_sample(uint8_t i, Vector3f& s) const {
+    if (i >= _samples_collected) {
+        return false;
+    }
+    s = _sample_buffer[i].delta_velocity / _sample_buffer[i].delta_time;
+    return true;
+}
+
+// returns truen and sample corrected with diag offdiag parameters as calculated by LSq estimation procedure
+// returns false if no correct parameter exists to be applied along with existing sample without corrections
+bool AccelCalibrator::get_sample_corrected(uint8_t i, Vector3f& s) const {
+    if (_status != ACCEL_CAL_SUCCESS || !get_sample(i,s)) {
+        return false;
+    }
+
+    Matrix3f M (
+        _param_struct.diag.x    , _param_struct.offdiag.x , _param_struct.offdiag.y,
+        _param_struct.offdiag.x , _param_struct.diag.y    , _param_struct.offdiag.z,
+        _param_struct.offdiag.y , _param_struct.offdiag.z , _param_struct.diag.z
+    );
+
+    s = M*(s+_param_struct.offset);
+
+    return true;
+}
+
+// checks if no new sample has been recieved for considerable amount of time
+void AccelCalibrator::check_for_timeout() {
+    static const uint32_t timeout = _conf_sample_time*2*1000 + 500;
```

Comment:
don't make timeout static. It's actually not just tidyness, its a bug, and on some platforms (such as QFlight with clang++) the constructor ordering can mean it will initialise as 0
Just make it "const uint32_t timeout"